drivers/sdhc/rcar_mmc.c

```diff
@@ -0,0 +1,1472 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_rcar_mmc
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/disk.h>
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/drivers/clock_control/renesas_cpg_mssr.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/cache.h>
+#include <soc.h>
+
+#include "rcar_mmc_registers.h"
+
+/**
+ * @note we don't need any locks here, because SDHC subsystem cares about it
+ */
+
+LOG_MODULE_REGISTER(rcar_mmc, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define DIV_ROUND_UP(n, d) (((n) + (d) - 1) / (d))
+
+/**
+ * @brief Renesas MMC host controller driver data
+ *
+ */
+struct mmc_rcar_data {
+	struct sdhc_io host_io;
+	struct sdhc_host_props props;
+
+	uint32_t ver : 8,
```

Comment:
yep, better memory usage

Reply:
> yep, better memory usage

`uint32_t ver : 8` may become `uint8_t ver`. 
`ddr_mode : 1` - `bool ddr_mode`. 

This may use couple of more bits, but this is much more readable. Also, accessing individual bits is much harder for CPU. You may save some bits in data section, but every field access will generate more CPU instructions, so your code section will increase. At least 16-32 bits  (1-2 instructions) more for every field access.